src/runners/sac.py

```diff
@@ -85,22 +94,33 @@ def run(self):
                         batch = self.replay_buffer.sample_batch()
                         self.agent.update(data=batch)
 
+            # Save every N episodes or when the current episode return is better than the best return
+            # Following the logic of now deprecated checkpoint_model
+            if ep_number % self.runner_config["save_every_nth_episode"] == 0:
+                save_path = f"{self.runner_config['model_save_dir']}/{self.exp_config['experiment_name']}/best_{self.exp_config['experiment_name']}_episode_{ep_number}.statedict"
+                self.agent.save_model(save_path)
+
+            elif ep_ret > self.best_ret:
+                self.best_ret = ep_ret
+                save_path = f"{self.runner_config['model_save_dir']}/{self.exp_config['experiment_name']}/best_{self.exp_config['experiment_name']}_episode_{ep_number}.statedict"
+                self.agent.save_model(save_path)
```

Comment:
There's gotta be a better way of doing this if statement bruh

Reply:
Cleaned it up! Thanks!